heron/scheduler-core/src/java/com/twitter/heron/scheduler/utils/SchedulerUtils.java

```diff
@@ -442,21 +442,24 @@ public static boolean setSchedulerLocation(
    * Encode the JVM options
    * <br> 1. Convert it into Base64 format
    * <br> 2. Add \" at the start and at the end
-   * <br> 3. replace "=" with "&amp;equals;"
+   * <br> 3. replace "=" with "(61)" and "&amp;equals;"
+   * Note that '=' is escaped in two different ways. '(61)' is the new escaping.
+   * '&amp;equals;' was the original way but it is not friendly to bash and is
+   * causing issues in some schedulers. It is still left there for backward compatibility
+   * reason
    *
    * @return encoded string
    */
   public static String encodeJavaOpts(String javaOpts) {
     String javaOptsBase64 = DatatypeConverter.printBase64Binary(
         javaOpts.getBytes(StandardCharsets.UTF_8));
-
-    return String.format("\"%s\"", javaOptsBase64.replace("=", "&equals;"));
+    return String.format("\"%s\"", javaOptsBase64.replace("=", "(61)").replace("=", "&equals;"));
```

Comment:
do we need .replace("=", "&equals;")) here?  All the "=" would already be replaced by the previous replace statement right?

Reply:
right. Technically it is not really necessary.

I left it here to be consistent with the decoding path and easier (just a little) for people to understand. In future we can remove both of them at the same time.

Reply:
gotcha can we just add some comments at the places this is done for future reference.  Thanks!

Reply:
or maybe a TODO for the future

Reply:
Thinking about it again. Now I don't think leaving the encoding part here is very helpful. Plus it looks wrong and could cause more confusion. I am removing this replace() call in the new update.

It is also easier to tell that the decode() part is for backward compatibility purpose only without the encode() part.